src/uu/wc/src/wc.rs

```diff
@@ -209,11 +210,13 @@ fn wc(files: Vec<String>, settings: &Settings) -> StdResult<(), i32> {
                     }
                 }
                 char_count += current_char_count;
-
                 if current_char_count > longest_line_length {
-                    // we subtract one here because `line.len()` includes the LF
-                    // matches GNU 'wc' behavior
-                    longest_line_length = current_char_count - 1;
+                    // -L is a GNU 'wc' extension so same behavior on LF
+                    if ends_lf {
+                        longest_line_length = current_char_count - 1;
+                    } else {
+                        longest_line_length = current_char_count;
+                    }
```

Comment:
Can you try with the "if let" syntax?
https://doc.rust-lang.org/reference/expressions/if-expr.html
```suggestion
                    longest_line_length = if ends_lf {
                        current_char_count - 1
                    } else {
                        current_char_count
                    }
```

More rust expectation :)
(I didn't test, so, not sure it works :)

Reply:
How about this?

```rust
longest_line_length = current_char_count
if ends_lf {
 longest_line_length = longest_line_length -1
}
```

Reply:
LLVM will optimize it to:

longest_line_length = current_char_count - (int) ends_lf;

I think the rust is:

```rust
longest_line_length = current_char_count - ends_lf.into();
```

Reply:
Cleaned up. I also eliminated the branch prediction above.